File: Erlang/microsecondTime.erl
%
%  Copyright © 2008, 2013  Russel Winder

-module(microsecondTime).
-export([microsecondTime/0]).

microsecondTime() ->
    {Megaseconds, Seconds, Microseconds} = erlang:now(),
    Megaseconds * 1000000 + Seconds  + Microseconds / 1000000.

File: C/microsecondTime.c
/*
 *  A C function to return the current microsecond timer.
 *
 *  Copyright © 2008, 2014  Russel Winder
 */

#include <sys/time.h>

#include "microsecondTime.h"

long long microsecondTime(void) {
  struct timeval value;
  gettimeofday(&value, 0);
  return ((long long)value.tv_sec) * 1000000ll + value.tv_usec;
}
